Another alignment flag used with wxEXPAND

when running the test suite on wx the last test fails with traceback:

<details>

```
======================================================================
FAIL: test_color_column (traitsui.tests.test_color_column.TestColorColumn)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/aayres/.edm/envs/traitsui-test-3.6-wx/lib/python3.6/site-packages/traitsui/tests/test_color_column.py", line 48, in test_color_column
    with store_exceptions_on_all_threads(), create_ui(data):
  File "/Users/aayres/.edm/envs/traitsui-test-3.6-wx/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/Users/aayres/.edm/envs/traitsui-test-3.6-wx/lib/python3.6/site-packages/traitsui/tests/_tools.py", line 177, in create_ui
    ui = object.edit_traits(**ui_kwargs)
  File "/Users/aayres/.edm/envs/traitsui-test-3.6-wx/lib/python3.6/site-packages/traits/has_traits.py", line 1678, in edit_traits
    args,
  File "/Users/aayres/.edm/envs/traitsui-test-3.6-wx/lib/python3.6/site-packages/traitsui/view.py", line 462, in ui
    ui.ui(parent, kind)
  File "/Users/aayres/.edm/envs/traitsui-test-3.6-wx/lib/python3.6/site-packages/traitsui/ui.py", line 246, in ui
    self.rebuild(self, parent)
  File "/Users/aayres/.edm/envs/traitsui-test-3.6-wx/lib/python3.6/site-packages/traitsui/wx/toolkit.py", line 125, in ui_live
    ui_live.ui_live(ui, parent)
  File "/Users/aayres/.edm/envs/traitsui-test-3.6-wx/lib/python3.6/site-packages/traitsui/wx/ui_live.py", line 52, in ui_live
    _ui_dialog(ui, parent, BaseDialog.NONMODAL)
  File "/Users/aayres/.edm/envs/traitsui-test-3.6-wx/lib/python3.6/site-packages/traitsui/wx/ui_live.py", line 88, in _ui_dialog
    BaseDialog.display_ui(ui, parent, style)
  File "/Users/aayres/.edm/envs/traitsui-test-3.6-wx/lib/python3.6/site-packages/traitsui/wx/ui_base.py", line 71, in display_ui
    ui.owner.init(ui, parent, style)
  File "/Users/aayres/.edm/envs/traitsui-test-3.6-wx/lib/python3.6/site-packages/traitsui/wx/ui_live.py", line 203, in init
    sw = panel(ui, window)
  File "/Users/aayres/.edm/envs/traitsui-test-3.6-wx/lib/python3.6/site-packages/traitsui/wx/ui_panel.py", line 272, in panel
    panel, content[0], ui
  File "/Users/aayres/.edm/envs/traitsui-test-3.6-wx/lib/python3.6/site-packages/traitsui/wx/ui_panel.py", line 431, in fill_panel_for_group
    panel, group, ui, suppress_label, is_dock_window, create_panel
  File "/Users/aayres/.edm/envs/traitsui-test-3.6-wx/lib/python3.6/site-packages/traitsui/wx/ui_panel.py", line 587, in __init__
    self.add_items(content, panel, self.sizer)
  File "/Users/aayres/.edm/envs/traitsui-test-3.6-wx/lib/python3.6/site-packages/traitsui/wx/ui_panel.py", line 909, in add_items
    editor.prepare(item_panel)
  File "/Users/aayres/.edm/envs/traitsui-test-3.6-wx/lib/python3.6/site-packages/traitsui/editor.py", line 240, in prepare
    self.init(parent)
  File "/Users/aayres/.edm/envs/traitsui-test-3.6-wx/lib/python3.6/site-packages/traitsui/wx/table_editor.py", line 200, in init
    self._create_toolbar(panel, sizer)
  File "/Users/aayres/.edm/envs/traitsui-test-3.6-wx/lib/python3.6/site-packages/traitsui/wx/table_editor.py", line 456, in _create_toolbar
    sizer.Add(tb_sizer, 0, wx.ALIGN_RIGHT | wx.EXPAND)
wx._core.wxAssertionError: C++ assertion "!(flags & (wxALIGN_RIGHT | wxALIGN_CENTRE_HORIZONTAL))" failed at /Users/robind/projects/bb2/dist-osx-py36/build/ext/wxWidgets/src/common/sizer.cpp(2089) in DoInsert(): Horizontal alignment flags are ignored with wxEXPAND


```

</details>

Another example of issue #828

--- a/traitsui/wx/table_editor.py
+++ b/traitsui/wx/table_editor.py
@@ -453,7 +453,7 @@ class TableEditor(Editor, BaseTableEditor):
                 tb_sizer.Add((1, 1), 1, wx.EXPAND)
                 tb_sizer.Add(toolbar.control, 0)
 
-            sizer.Add(tb_sizer, 0, wx.ALIGN_RIGHT | wx.EXPAND)
+            sizer.Add(tb_sizer, 0, wx.EXPAND)
 
     def dispose(self):
         """ Disposes of the contents of an editor.

